### PR TITLE
[wt] Update to 4.10.4

### PR DESCRIPTION
**Pull request by MonicaLiu0311**
Update `wt` to 4.10.4. 

* Remove outdated `0002-link-glew.patch` and replace outdated PR #33369.

* Fix the following errors:
```
1> [CMake] CMake Error at F:/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package): 
1> [CMake] Could not find a package configuration file provided by "wt" with any of 
1> [CMake] the following names: 
1> [CMake] 
1> [CMake] wtConfig.cmake 
1> [CMake] wt-config.cmake 
1> [CMake] 
1> [CMake] Add the installation prefix of "wt" to CMAKE_PREFIX_PATH or set "wt_DIR" to 
1> [CMake] a directory containing one of the above files. If "wt" provides a separate 
1> [CMake] development package or SDK, be sure it has been installed. 
1> [CMake] Call Stack (most recent call first): 
1> [CMake] CMakeUsage/CMakeLists.txt:17 (find_package)
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```
The usage test passed on `x64-windows` (header files found):
```
wt provides CMake targets:

  # this is heuristically generated, and may not be correct
  find_package(wt CONFIG REQUIRED)
  target_link_libraries(main PRIVATE Wt::Wt Wt::HTTP Wt::Test Wt::Isapi)
```

### CODE REVIEW

File: ports/wt/vcpkg.json [hunk 1]
```diff
@@ -75,10 +76,22 @@
         "openssl"
       ]
     },
+    "pango": {
+      "description": "For improved font support in the WPdfImage and WRasterImage paint devices.",
+      "dependencies": [
+        {
+          "name": "pango",
+          "platform": "!windows"
+        }
```
Comment (dg0yt) [inline ports/wt/vcpkg.json:85]: Can there be pango support without pango?

File: ports/wt/vcpkg.json [hunk 2]
```diff
@@ -66,7 +64,10 @@
     "graphicsmagick": {
       "description": "WRasterImage support on non-Windows based on GraphicsMagick",
       "dependencies": [
-        "graphicsmagick"
+        {
+          "name": "graphicsmagick",
+          "platform": "!(android | ios)"
+        }
```
Comment (dg0yt) [inline ports/wt/vcpkg.json:70]: Can there be ...

**Review by dg0yt — COMMENTED** [review]
Did you actually want to restrict some features' `supports`?
Is this needed, or is this already covered by the dependencies' `supports`?

File: ports/wt/0007-Wttest.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index eb52234..1bb0fa1 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -607,7 +607,7 @@ IF(ENABLE_LIBWTTEST)
+     VERSION ${VERSION_SERIES}.${VERSION_MAJOR}.${VERSION_MINOR}
+     DEBUG_POSTFIX ${DEBUG_LIB_POSTFIX}
+   )
+-ELSE(ENABLE_LIBWTTEST)
++ELSE(NOT ENABLE_LIBWTTEST)
```
Comment (dg0yt) [inline ports/wt/0007-Wttest.patch:10]: What is this meant to do? 
Note that this is an `ELSE`, not an `ELSEIF`. Old cmake style, today one would just use `else()`. But style is not a reason for patching.

File: ports/wt/portfile.cmake [hunk 4]
```diff
@@ -70,12 +70,15 @@ vcpkg_cmake_configure(
...
 
         -DUSE_SYSTEM_SQLITE3=ON
-        -DUSE_SYSTEM_GLEW=ON
 
         -DCMAKE_INSTALL_DIR=share
         # see https://redmine.webtoolkit.eu/issues/9646
         -DWTHTTP_CONFIGURATION=
         -DCONFIGURATION=
+        
+        "-DCMAKE_PREFIX_PATH=${CURRENT_INSTALLED_DIR}"
```
Comment (dg0yt) [inline ports/wt/portfile.cmake:79]: What is this meant to do?

Comment (MonicaLiu0311) [reply ports/wt/portfile.cmake:79]: Fix the following errors:
```
1> [CMake] CMake Error at F:/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package): 
1> [CMake] Could not find a package configuration file provided by "wt" with any of 
1> [CMake] the following names: 
1> [CMake] 
1> [CMake] wtConfig.cmake 
1> [CMake] wt-config.cmake 
1> [CMake] 
1> [CMake] Add the installation prefix of "wt" to CMAKE_PREFIX_PATH or set "wt_DIR" to 
1> [CMake] a directory containing one of the above files. If "wt" provides a separate 
1> [CMake] development package or SDK, be sure it has been installed. 
1> [CMake] Call Stack (most recent call first): 
1> [CMake] CMakeUsage/CMakeLists.txt:17 (find_package)
```

https://github.com/emweb/wt/blob/4.10.4/CMakeLists.txt#L142-L148

File: ports/wt/0007-Wttest.patch (hunk 3 above)
Comment (MonicaLiu0311) [reply ports/wt/0007-Wttest.patch]: I misunderstood, and it’s okay to judge this way. Has been modified.

File: ports/wt/portfile.cmake (hunk 4 above)
Comment (dg0yt) [reply ports/wt/portfile.cmake:78]: So you don't want to and need to override the vcpkg-provided CMAKE_PREFIX_PATH, but set input variable USERLIB_PREFIX instead.

https://github.com/emweb/wt/blob/c10dbd54e3151fa48851f30aff9ad0a582f1a7b6/CMakeLists.txt#L156-L157

```suggestion
        "-DUSERLIB_PREFIX=${CURRENT_INSTALLED_DIR}"
```

And maybe this still isn't safe for debug builds.

**Pull request merged**